src/llamafactory/chat/hf_engine.py

```diff
@@ -86,22 +92,37 @@ def _process_args(
             and image is not None
             and not hasattr(processor, "image_seq_length")
             and template.image_token not in messages[0]["content"]
-        ):  # llava-like models
+            and model_args.visual_inputs_type in ["vision_tower","phi3v_like"]
```

Comment:
Every commit should be formatted by using `make style && make quality` to avoid such codes